vmware/ngc/NGC-Register/src/main/java/org/opensds/vmware/ngc/utils/XmlUtils.java

```diff
@@ -0,0 +1,350 @@
+            event.setEventLocal(local);
+            eventInfoList.add(event);
+        }
+        return eventInfoList;
+    }
+
+    public VCenterInfo xmlReadVcenterInfo() {
+        return xmlReadVcenterInfo(VCENTER_CONFIG);
+    }
+
+    public VCenterInfo xmlReadVcenterInfo(String path) {
+        VCenterInfo vcInfo = new VCenterInfo();
+        try {
+            if (path == null) {
+                path = VCENTER_CONFIG;
+            }
+            Document doc = loadConigXml(path);
+            Element root = doc.getRootElement();
+            Iterator< ? > it = root.elementIterator();
+
+            while (it.hasNext()) {
+                Element vcElement = (Element) it.next();
+                String vcip = vcElement.attributeValue("vcip");
+                String vcname = vcElement.attributeValue("vcname");
+                String vcpwd = vcElement.attributeValue("vcpwd");
+                String flag = vcElement.attributeValue("registerflag");
+                vcInfo.setvCenterIp(vcip);
+                vcInfo.setvCenterUser(vcname);
+                vcInfo.setvCenterPassword(vcpwd);
+                vcInfo.setRegisterFlag(flag);
+            }
+
+        } catch (Exception ex) {
+            _logger.error("Read XML error, ex : " + ex.getMessage());
+        }
+        return vcInfo;
+    }
+
+    public ErrorCode xmlWriteVcenterInfo(VCenterInfo vc) {
+        Document doc = loadConigXml(VCENTER_CONFIG);
+        Element root = doc.getRootElement();
+        Iterator< ? > it = root.elementIterator();
+        while (it.hasNext())
+        {
+            Element vcElement = (Element) it.next();
+            vcElement.attribute("vcip").setText(vc.getvCenterIp());
+            vcElement.attribute("vcname").setText(vc.getvCenterUser());
+            vcElement.attribute("vcpwd").setText(vc.getvCenterPassword());
+            vcElement.attribute("registerflag").setText(vc.getRegisterFlag());
+        }
+        return writeToXML(VCENTER_CONFIG, doc);
+    }
+
+    private ErrorCode writeToXML(String path, Document doc) {
+        if (OperationSystemUtils.isFreeDiskSpace()) {
+            return ErrorCode.NO_SPACE;
+        }
+        OutputStream outputStream = null;
+        Writer writer = null;
+        XMLWriter xmlWriter = null;
+        try {
+            outputStream = new FileOutputStream(path);
+            writer = new OutputStreamWriter(outputStream, "UTF-8");
+            xmlWriter = new XMLWriter(writer);
+            xmlWriter.write(doc);
+            xmlWriter.flush();
+            return ErrorCode.SUCCESS;
+        } catch (IOException e) {
+            _logger.error("Writer xml error ;" + e.toString());
+            return ErrorCode.WRITE_CONFIG_FAILED;
+        } finally {
+            if (xmlWriter != null) {
+                try {
+                    xmlWriter.close();
+                }
+                catch (IOException e) {
+                    _logger.error("XMLWriter close error" + e.toString());
```

Comment:
why no error code?

Reply:
Since it is in finally block, we are not returning any error code.